src/Server.php

```diff
@@ -319,6 +325,26 @@ public function getCacheWithFileExtensions()
         return $this->cacheWithFileExtensions;
     }
 
+    /**
+     * Set a custom cachePathCallable.
+     *
+     * @param callable|null $cachePathCallable The custom cache path callable. It receives the same arguments as @see getCachePath
```

Comment:
I would prefer restricting this to `Closure` instead of `callable` and add `Closure` type for the argument in method signature. `callable` allows a bazillions things making `callable` type checking expensive. 

Users can always use `Closure::fromCallable()` for the method call if they really need to use other callable types.